elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -431,6 +433,69 @@ export abstract class BaseTransaction {
 		this._id = getId(this.getBytes());
 	}
 
+	public signAll(
+		networkIdentifier: string,
+		senderPassphrase?: string,
+		passphrases?: ReadonlyArray<string>,
+		keys?: {
+			readonly mandatoryKeys: Array<Readonly<string>>;
+			readonly optionalKeys: Array<Readonly<string>>;
+			readonly numberOfSignatures: number;
+		},
+	): void {
+		// Set network identifier if it was previously not set in the transaction
+		if (!this._networkIdentifier) {
+			this._networkIdentifier = networkIdentifier;
+		}
+
+		if (!this.signatures) {
+			this.signatures = [];
```

Comment:
I think `signatures` is initialized in the constructor?

Reply:
Fixed